### PR TITLE
[SCRIPTS] scenarios coverage

### PR DESCRIPTION
**Pull request by nikohosn**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,6 +1,7 @@
 NODE_URL=ws://node:9944
-SMART_CONTRACT_ADDRESS=SMART_CONTRACT_ADDRESS
-APP_WALLET_JSON=APP_WALLET_JSON
-APP_WALLET_PASSPHRASE=APP_WALLET_PASSPHRASE
+SMART_CONTRACT_ADDRESS=5CgGJhgZdUHpsC6nHy3Y1iVpvtbW7EhBKvuAXE5WFsbmBVFk
```
Comment (evgeny-s) [inline .env:2]: Please revert all changes from this file

File: README.md [hunk 2]
```diff
@@ -27,21 +27,28 @@ APP_WALLET_PASSPHRASE
 ```
 #### 5. Run tests
 
+**Run Build Command**
```
Comment (evgeny-s) [inline README.md:30]: Why do we need this? It will build the image anyway.

File: src/substrate-service.ts [hunk 3]
```diff
@@ -66,6 +66,37 @@ export class SubstrateService {
     });
   }
 
+  public async issueAssetToUserBatch(destinationAccountPublicKey: string, amount: string, key_2: string, key_3: string): Promise<IssueAssetResponse> {
```
Comment (evgeny-s) [inline src/substrate-service.ts:69]: Let this function will accept the array of keys, so it is flexible on amount if items in batch to process

**Review by evgeny-s — CHANGES_REQUESTED** [review]
Please check comments

File: .env (hunk 1 above)
Comment (AndreiNavoichyk) [reply .env:2]: @nikohosn 
This is template. We should not change here anything and push

File: .env [hunk 4]
```diff
@@ -1,6 +1,7 @@
...
+APP_WALLET_JSON={"address":"5EeJGvXetxrb2N33YdxZApd99i92uznpXaUdW8Nprfa2rqTS","encoded":"s6N+ZdLoXGdjCYmN3/jxfLwDTi1//CPeshwsSFrBXOkAgAAAAQAAAAgAAADz5ze2/IQsuBv2yhYP4gejynu9t8fIW+C5C8hNjj93NLCNSf2TZ/NLncVOyp+1BJe7yHW2MCy2AaFb14ZfbS7uyYY9qHyXbU1IOBiWZUX3qBCv2UhksEVrt67N3dH3b5maa9Uj/n9ywvPSqnTQREP5jt4drnWdHzp5OvTpGvxyghmKo9bUAmW8CL0j3PRVyldCkReFhms2iC1hhYbA","encoding":{"content":["pkcs8","sr25519"],"type":["scrypt","xsalsa20-poly1305"],"version":"3"},"meta":{"name":"cere","whenCreated":1611176126091,"isHidden":false}}
+APP_WALLET_PASSPHRASE=Beaufort99$
 AMOUNT=1000000000000000
```
Comment (AndreiNavoichyk) [inline .env:5]: Let's apply `SCENARIO_NUMBER` prefix and for `SCENARIO_1` also.

File: src/scenarios/scenarios_2.ts [hunk 5]
```diff
@@ -3,12 +3,47 @@ import {ScenarioInterface} from "./scenario-interface";
 
 export class Scenarios_2 implements ScenarioInterface {
   async run() {
-    /*
+        /*
```
Comment (AndreiNavoichyk) [inline src/scenarios/scenarios_2.ts:6]: Let's add a short scenarios description in README and remove the comment from here.

File: src/scenarios/scenarios_3.ts [hunk 6]
```diff
@@ -1,12 +1,52 @@
...
 import {ScenarioInterface} from "./scenario-interface";
+import API from '@polkadot/api';
 
 export class Scenarios_3 implements ScenarioInterface {
   async run() {
     /*
        TODO: Implement Scenario 3 [Batch processing]:
          * Create 3 user wallets
-         * Sends assets to those wallets using batch transaction using link: https://github.com/Cerebellum-Network/private-standalone-network-node/blob/feature/fee-abstraction-doc/docs/batch_processing.md#2-using-polkadot-utilitybatch-method
+         * Sends assets to those wallets using batch transaction using link: https://github.com/Cerebellum-Network/private-standalone-network-node/blob/dev/docs/batch_processing.md
```
Comment (AndreiNavoichyk) [inline src/scenarios/scenarios_3.ts:10]: Let's add a short scenarios description in README and remove the comment from here.

File: src/scenarios/scenarios_3.ts [hunk 7]
```diff
@@ -1,12 +1,52 @@
...
+
+    console.log('[MAIN] Generating wallet...');
+    const banana = service.generateWallet();
+    console.log(`[MAIN] Wallet has been generated successfully. Value is ${JSON.stringify(banana)}`);
+
+    // Batch transactions
+    console.log(`[MAIN] Issuing assets to the user...`);
+    const result = await service.issueAssetToUserBatch(
+      apple.publicKey,
+      process.env.AMOUNT,
```
Comment (AndreiNavoichyk) [inline src/scenarios/scenarios_3.ts:35]: Let's use `SCENARIO_3_AMOUN` here.

File: src/substrate-service.ts [hunk 8]
```diff
@@ -66,6 +66,37 @@ export class SubstrateService {
...
+    const txs = [
+      await this.substrateApi.tx.balances.transfer(
```
Comment (AndreiNavoichyk) [inline src/substrate-service.ts:71]: Do we need `await` here?

File: src/substrate-service.ts [hunk 9]
```diff
@@ -86,4 +117,27 @@ export class SubstrateService {
...
     throw new Error('Something went wrong!');
   }
+
+  public async getFeeEstimate(destinationAccountPublicKey: string, amount: string): Promise<any> {
+    let info = await this.substrateApi.tx.balances.transfer(
+      destinationAccountPublicKey,
+      amount
+    ).paymentInfo(this.appKeyring)
+    
+    console.log(`
```
Comment (AndreiNavoichyk) [inline src/substrate-service.ts:127]: Do we need this log?

File: src/substrate-service.ts (hunk 3 above)
Comment (AndreiNavoichyk) [inline src/substrate-service.ts:69]: @nikohosn 
Please, take a look to Cere's code guidelines. There you can find code conventions for variables, classes and etc.

File: .env (hunk 1 above)
Comment (nikohosn) [reply .env:2]: Removed

File: README.md (hunk 2 above)
Comment (nikohosn) [reply README.md:30]: removed

File: .env (hunk 4 above)
Comment (nikohosn) [reply .env:5]: changed this to only use one standard AMOUNT. No need to have multiple amounts

File: src/substrate-service.ts (hunk 3 above)
Comment (nikohosn) [reply src/substrate-service.ts]: Resolved and changed per request

File: src/scenarios/scenarios_3.ts (hunk 6 above)
Comment (nikohosn) [reply src/scenarios/scenarios_3.ts]: Added to readme

File: src/scenarios/scenarios_2.ts (hunk 5 above)
Comment (nikohosn) [reply src/scenarios/scenarios_2.ts]: Added to readme

File: src/scenarios/scenarios_3.ts (hunk 7 above)
Comment (nikohosn) [reply src/scenarios/scenarios_3.ts:30]: changed this to only use one standard AMOUNT. No need to have multiple amounts

File: src/substrate-service.ts (hunk 8 above)
Comment (nikohosn) [reply src/substrate-service.ts]: Fixed

File: src/substrate-service.ts (hunk 9 above)
Comment (nikohosn) [reply src/substrate-service.ts]: Removed

File: src/substrate-service.ts (hunk 3 above)
Comment (nikohosn) [reply src/substrate-service.ts]: Understood, resolved, and removed the ambiguous namings

File: src/scenarios/scenarios_3.ts [hunk 10]
```diff
@@ -1,12 +1,46 @@
...
+
+    const appleBalance = await service.getBalance(alice.publicKey);
+    console.log(`[MAIN] Asset balance: ${JSON.stringify(appleBalance)}`);
+
+    const orangeBalance = await service.getBalance(jim.publicKey);
+    console.log(`[MAIN] Asset balance: ${JSON.stringify(orangeBalance)}`);
+
+    const bananaBalance = await service.getBalance(bob.publicKey);
+    console.log(`[MAIN] Asset balance: ${JSON.stringify(bananaBalance)}`);
+
```
Comment (AndreiNavoichyk) [inline src/scenarios/scenarios_3.ts:43]: Please, remove empty lines.

**Review by AndreiNavoichyk — APPROVED** [review]

File: .env [hunk 11]
```diff
@@ -1,6 +1,6 @@
 NODE_URL=ws://node:9944
-SMART_CONTRACT_ADDRESS=SMART_CONTRACT_ADDRESS
```
Comment (evgeny-s) [inline .env:2]: @nikohosn , please keep the same values like:
```
SMART_CONTRACT_ADDRESS=SMART_CONTRACT_ADDRESS
APP_WALLET_JSON=APP_WALLET_JSON
APP_WALLET_PASSPHRASE=APP_WALLET_PASSPHRASE
```

File: src/scenarios/scenarios_3.ts [hunk 12]
```diff
@@ -1,12 +1,46 @@
 import {SubstrateService} from "../substrate-service";
 import {ScenarioInterface} from "./scenario-interface";
+import API from '@polkadot/api';
```
Comment (evgeny-s) [inline src/scenarios/scenarios_3.ts:3]: Why do you need this here?

File: src/substrate-service.ts [hunk 13]
```diff
@@ -66,6 +66,32 @@ export class SubstrateService {
     });
   }
 
+  public async issueAssetToUserBatch(amount: string, wallets: Array<string>): Promise<IssueAssetResponse> {
+    let i
+    let txs = []
+    for (i = 0; i < wallets.length; i++) {
```
Comment (evgeny-s) [inline src/substrate-service.ts:72]: Let's initialize `i` inside the for block:
```
for (let i = 0; ...)
```

File: src/substrate-service.ts [hunk 14]
```diff
@@ -66,6 +66,32 @@ export class SubstrateService {
     });
   }
 
+  public async issueAssetToUserBatch(amount: string, wallets: Array<string>): Promise<IssueAssetResponse> {
+    let i
+    let txs = []
```
Comment (evgeny-s) [inline src/substrate-service.ts:71]: Let's put a semicolon at the end of the lines as well.

**Review by evgeny-s — CHANGES_REQUESTED** [review]
Please check comments

File: src/substrate-service.ts (hunk 14 above)
Comment (AndreiNavoichyk) [reply src/substrate-service.ts:71]: @nikohosn 
Be sure you've installed prettier correctly in you IDE.

**Pull request merged**